src/openeo_test_suite/tests/general/test_compliance.py

```diff
@@ -0,0 +1,1494 @@
+        )
+    )
+
+    # TESTING
+
+    id = str(uuid.uuid4())
+    prepared_endpoint_path = f"{endpoint_path}/{id}"
+    fail_log += conformance_util.test_endpoint(
+        base_url=base_url,
+        endpoint_path=prepared_endpoint_path,
+        test_name=f"{test_name} {id}",
+        spec=spec,
+        payload=payload,
+        bearer_token=bearer_token,
+        method="PUT",
+        expected_status_codes=range(400, 501),
+    )
+
+    # CLEANUP
+
+    assert fail_log == ""
+
+
+def test_empty_POST_jobs(base_url: str, spec: Spec, bearer_token: str):
+    """
+    setup: prepare empty payloads
+    testing: test posting empty payloads endpoint
+    cleanup: None
+    """
+    # SETUP
+    endpoint_path = "jobs"
+    test_name = "Creates a new batch processing task (NEGATIVE)"
+    directory_path = conformance_util.get_examples_path()
+    examples_directory = "empty_payload"
+
+    payload = next(
+        conformance_util.load_payloads_from_directory(
+            directory_path=f"{directory_path}/{examples_directory}"
+        )
+    )
+
+    # TESTING
+    fail_log = conformance_util.test_endpoint(
+        base_url=base_url,
+        endpoint_path=endpoint_path,
+        test_name=test_name,
+        spec=spec,
+        bearer_token=bearer_token,
+        payload=payload,
+        method="POST",
+        expected_status_codes=range(400, 501),
+    )
+
+    # CLEANUP
+    assert fail_log == ""
+
+
+def test_empty_PATCH_jobs_job_id(base_url: str, spec: Spec, bearer_token: str):
+    """
+    SETUP: POST jobs, prepare payloads
+    TESTING: PATCH jobs with empty payload
+    CLEANUP: DELETE jobs
+    """
+
+    # SETUP
+    fail_log = ""
+    endpoint_path = "jobs"
+    test_name = "Negative Modify a batch job"
+
+    created_batch_job_ids = conformance_util.post_jobs(
+        base_url=base_url, bearer_token=bearer_token
+    )
+
+    directory_path = conformance_util.get_examples_path()
+    examples_directory = "empty_payload"
+
+    payload = next(
+        conformance_util.load_payloads_from_directory(
+            directory_path=f"{directory_path}/{examples_directory}"
+        )
+    )
+
+    # TESTING
+    for job_id in created_batch_job_ids:
+        prepared_endpoint_path = f"{endpoint_path}/{job_id}"
+        fail_log += conformance_util.test_endpoint(
+            base_url=base_url,
+            endpoint_path=prepared_endpoint_path,
+            test_name=f"{test_name} {job_id}",
+            spec=spec,
+            bearer_token=bearer_token,
+            payload=payload,
+            method="PATCH",
+            expected_status_codes=204,
+        )
+
+    # CLEANUP
+    conformance_util.delete_id_resource(
+        base_url=base_url,
+        endpoint_path=endpoint_path,
+        bearer_token=bearer_token,
+        ids=created_batch_job_ids,
+    )
+
+    assert fail_log == ""
+
+
+@pytest.mark.skip(
+    reason="Takes too long for normal test runs, this decorator can be removed when fully testing is relevant"
```

Comment:
This should probably be customizable through a CLI parameter? Potentially to be aligned with Micheles tests in the other WP?

Reply:
Indeed. I think it makes sense to to mark long running tests with e.g.

```
@pytest.mark.longrunning
def test_...
```

and then these tests can be skipped with pytest built-in marker based test selction feature . E.g. skip longrunning tests:
```
pytest .... -m "not longrunning"
```

also see https://docs.pytest.org/en/7.1.x/example/markers.html

Reply:
This is done.